pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -212,32 +212,36 @@ func toKubeRuntimeStatus(status *runtimeapi.RuntimeStatus) *kubecontainer.Runtim
 	return &kubecontainer.RuntimeStatus{Conditions: conditions}
 }
 
-func fieldProfile(scmp *v1.SeccompProfile, profileRootPath string, fallbackToRuntimeDefault bool) string {
+func fieldProfile(scmp *v1.SeccompProfile, profileRootPath string, fallbackToRuntimeDefault bool) (string, error) {
 	if scmp == nil {
 		if fallbackToRuntimeDefault {
-			return v1.SeccompProfileRuntimeDefault
+			return v1.SeccompProfileRuntimeDefault, nil
 		}
-		return ""
+		return "", nil
 	}
 	if scmp.Type == v1.SeccompProfileTypeRuntimeDefault {
-		return v1.SeccompProfileRuntimeDefault
+		return v1.SeccompProfileRuntimeDefault, nil
 	}
-	if scmp.Type == v1.SeccompProfileTypeLocalhost && scmp.LocalhostProfile != nil && len(*scmp.LocalhostProfile) > 0 {
-		fname := filepath.Join(profileRootPath, *scmp.LocalhostProfile)
-		return v1.SeccompLocalhostProfileNamePrefix + fname
+	if scmp.Type == v1.SeccompProfileTypeLocalhost {
+		if scmp.LocalhostProfile != nil && len(*scmp.LocalhostProfile) > 0 {
+			fname := filepath.Join(profileRootPath, *scmp.LocalhostProfile)
+			return v1.SeccompLocalhostProfileNamePrefix + fname, nil
+		} else {
+			return "", fmt.Errorf("localhostProfile must be set if seccompProfile type is Localhost.")
+		}
 	}
 	if scmp.Type == v1.SeccompProfileTypeUnconfined {
-		return v1.SeccompProfileNameUnconfined
+		return v1.SeccompProfileNameUnconfined, nil
 	}
```

Comment:
I think this would read a lot better as a switch

```go
switch scmp.Type {
  case v1.SeccompProfileTypeRuntimeDefault:
    return v1.SeccompProfileTypeRuntimeDefault, nil

  case v1.SeccompProfileTypeLocalhost:
    if scmp.LocalhostProfile != nil && len(*scmp.LocalhostProfile) > 0 {
      fname := filepath.Join(profileRootPath, *scmp.LocalhostProfile)
      return v1.SeccompLocalhostProfileNamePrefix + fname, nil
    } else {
      return "", fmt.Errorf("localhostProfile must be set if seccompProfile type is Localhost.")
    }
  
  case v1.SeccompProfileTypeUnconfined:
    return v1.SeccompProfileNameUnconfined, nil

  default:
    if fallbackToRuntimeDefault {
      return v1.SeccompProfileRuntimeDefault, nil
    }
    return "", nil
}
```

Reply:
Thanks, I agree that does look more readable. I was trying to avoid making too many changes in my first PR here but this was straightforward enough to do and tests still pass :)